### ISSUE TITLE
Slow seeking when streaming from YouTube

### ISSUE THREAD

**Issue by low-batt** [iina/iina#4522]
**System and IINA version:**

- macOS unknown
- IINA 1.3.2

**Expected behavior:**
Seeking with the arrow keys is fast.

**Actual behavior:**
In closed issue #4153 @YU0211 reported:
> i tried to play youtube video by pasting link to IINA , in that circumstance I have to wait about 3 seconds every time i press left or right or scrolling .The internet speed is about 300M , so I think the issue may not comes from the internet.

As the issue with Bluetooth audio devices has been fixed in IINA 1.3.2 I am opening up a new issue for this report.

**Steps to reproduce:**
Stream a video from YouTube and seek using the arrow keys.

- [ ] MPV does not have this problem.

Unknown.

**How often does this happen?**
Unknown.

**Comment by low-batt** [comment]
I ran a test streaming [Big Buck Bunny 60fps 4K](https://www.youtube.com/watch?v=aqz-KE-bpKQ) from YouTube with audio directed to a Bluetooth speaker. Seeking with the arrow keys worked pretty well. I did encounter pauses when the seek went outside of the buffered parts of the video and IINA was waiting on YouTube.

Having a reasonable speed connection from your ISP is important. _However_ a lot of issues with streaming from YouTube have been traced to YouTube throttling traffic.

The version of [yt-dlp](https://github.com/yt-dlp/yt-dlp) being used is critical. I ran this test with the version embedded in IINA 1.3.2:

```text
low-batt@gag ~$ /Applications/IINA.app/Contents/MacOS/youtube-dl --version
2023.06.22
low-batt@gag ~$ 
```

One way to use the latest version of `yt-dlp` is to install it using [brew](https://brew.sh/) and keep it up to date using that package manager. This shows me checking the version of `yt-dlp` installed by `brew` using [Terminal](https://support.apple.com/guide/terminal/welcome/mac):
```text
low-batt@gag ~$ which yt-dlp
/opt/homebrew/bin/yt-dlp
low-batt@gag ~$ yt-dlp --version
2023.07.06
low-batt@gag ~$ 
```

To configure IINA to use the `yt-dlp` version installed by `brew` follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Network`
- For the `Custom youtube-dl path` setting specify the full pathname to the directory containing `yt-dlp` 
- **Restart IINA** to activate the new settings

Where `brew` installed `yt-dlp` can be found using the `prefix` option as shown here: 
```text
low-batt@gag ~$ brew --prefix yt-dlp
/opt/homebrew/opt/yt-dlp
low-batt@gag ~$ 
```

I configured IINA to use the `brew` installed version:

![issue-slow-youtube](https://github.com/iina/iina/assets/86170219/bfcddfea-3187-4916-abb0-14ccb2e05ac6)

And tested again. For the issue at hand I did not see a difference in seeking speed. _However_ if you are experiencing issues with streaming from YouTube it is always desirable to try the latest `yt-dlp` and see if it fixes the problem.

So far I'm thinking what you are experiencing is YouTube behavior.
